### PR TITLE
Adddorgbr

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by kortschak** [comment]
This consistent green is lovely.

File: native/dorgbr.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+// Q is of order m. If m >= k, then Dorgbr returns the first n columns of Q
+// where m >= n >= k. If m < k, then Dorgbr returns Q as an m×m matrix.
+//
+// If vect == lapack.ApplyP, then A is assumed to have been a k×n matrix, and
+// P^T is of order n. If k < n, then Dorgbr returns the first m rows of P^T,
+// where n >= m >= k. If k >= n, then Dorgbr returns P^T as an n×n matrix.
+func (impl Implementation) Dorgbr(vect lapack.DecompUpdate, m, n, k int, a []float64, lda int, tau, work []float64, lwork int) {
+	mn := min(m, n)
+	wantq := vect == lapack.ApplyQ
+	if wantq {
```
Comment (kortschak) [inline native/dorgbr.go]: The fortran equivalent of this is awful.

File: native/dorgbr.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		checkMatrix(m, k, a, lda)
+	} else {
+		checkMatrix(k, n, a, lda)
+	}
+	work[0] = 1
+	if wantq {
+		if m >= k {
+			impl.Dorgqr(m, n, k, a, lda, tau, work, -1)
+		} else {
```
Comment (kortschak) [inline native/dorgbr.go]: Condense this and below to `else if`?

**Comment by btracey** [comment]
Shhhh don't jinx it.

File: native/dorgbr.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+		return
+	}
+	if len(work) < lwork {
+		panic(badWork)
+	}
+	if lwork < mn {
+		panic(badWork)
+	}
+	if m == 0 || n == 0 {
+		return
```
Comment (kortschak) [inline native/dorgbr.go]: Set `work[0] = 1` before return?

File: native/dorgbr.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	}
+	if wantq {
```
Comment (kortschak) [inline native/dorgbr.go]: The comments in the fortran here are worth transcribing.

**Comment by kortschak** [comment]
LGTM, but as always please wait for @vladimir-ch.

File: cgo/lapack.go [hunk 5]
```diff
@@ -542,6 +543,47 @@ func (impl Implementation) Dgetrs(trans blas.Transpose, n, nrhs int, a []float64
 	clapack.Dgetrs(trans, n, nrhs, a, lda, ipiv32, b, ldb)
 }
 
+// Dorgbr generates one of the matrices Q or P^T computed by Dgebrd
+// computed from the decomposition Dgebrd. See Dgebd2 for the description of
```
Comment (vladimir-ch) [inline cgo/lapack.go]: This seems to be an editing leftover.

File: native/dorgbr.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import "github.com/gonum/lapack"
+
+// Dorgbr generates one of the matrices Q or P^T computed by Dgebrd
+// computed from the decomposition Dgebrd. See Dgebd2 for the description of
```
Comment (vladimir-ch) [inline native/dorgbr.go]: Same here

**Comment by vladimir-ch** [comment]
LGTM

File: cgo/lapack.go (hunk 5 above)
Comment (btracey) [inline cgo/lapack.go]: No, that one's correct.

File: native/dorgbr.go (hunk 6 above)
Comment (btracey) [inline native/dorgbr.go]: Nope, the descriptions are in Dgebd2

File: native/dorgbr.go (hunk 1 above)
Comment (btracey) [inline native/dorgbr.go]: Yea...

File: native/dorgbr.go (hunk 2 above)
Comment (btracey) [inline native/dorgbr.go]: Done.

File: native/dorgbr.go (hunk 3 above)
Comment (btracey) [inline native/dorgbr.go]: Done.

File: native/dorgbr.go (hunk 4 above)
Comment (btracey) [inline native/dorgbr.go]: Done.

File: cgo/lapack.go (hunk 5 above)
Comment (vladimir-ch) [inline cgo/lapack.go]: I was referring to the `... computed by Dgebrd computed from the decomposition Dgebrd. ...` part.

Comment (btracey) [inline cgo/lapack.go]: Ah, sorry, I'll fix when I get a chance. Thanks!

**Pull request merged**